plugins/flutter_plugins/url_launcher_fde/linux/url_launcher_plugin.cc

```diff
@@ -97,6 +103,30 @@ void UrlLauncherPlugin::HandleMethodCall(
       return;
     }
     result->Success();
+  // Handle canLaunch
+  } else if (method_call.method_name().compare("canLaunch") == 0) {
+    std::string url;
+    if (method_call.arguments() && method_call.arguments()->IsMap()) {
+      const EncodableMap &arguments = method_call.arguments()->MapValue();
+      auto url_it = arguments.find(EncodableValue("url"));
+      if (url_it != arguments.end()) {
+        url = url_it->second.StringValue();
+      }
+    }
+    if (url.empty()) {
+      result->Error("argument_error", "No URL provided");
+      return;
+    }
+
+    if (startsWith(url, "https://") || (startsWith(url, "http://")) || (startsWith(url, "ftp://"))) {
```

Comment:
Since the scheme ends at the `:`, you could just use `"https:"`, `"http:"`, etc.